r-package/grf/R/boosted_regression_forest.R

```diff
@@ -0,0 +1,229 @@
+#'                      with size smaller than min.node.size can occur, as in the original randomForest package.
+#' @param honesty Whether to use honest splitting (i.e., sub-sample splitting).
+#' @param honesty.fraction The fraction of data that will be used for determining splits if honesty = TRUE. Corresponds
+#'                         to set J1 in the notation of the paper. When using the defaults (honesty = TRUE and
+#'                         honesty.fraction = NULL), half of the data will be used for determining splits
+#' @param ci.group.size The forest will grow ci.group.size trees on each subsample.
+#'                      In order to provide confidence intervals, ci.group.size must
+#'                      be at least 2.
+#' @param alpha A tuning parameter that controls the maximum imbalance of a split.
+#' @param imbalance.penalty A tuning parameter that controls how harshly imbalanced splits are penalized.
+#' @param seed The seed for the C++ random number generator.
+#' @param clusters Vector of integers or factors specifying which cluster each observation corresponds to.
+#' @param samples_per_cluster If sampling by cluster, the number of observations to be sampled from
+#'                            each cluster when training a tree. If NULL, we set samples_per_cluster to the size
+#'                            of the smallest cluster. If some clusters are smaller than samples_per_cluster,
+#'                            the whole cluster is used every time the cluster is drawn. Note that
+#'                            clusters with less than samples_per_cluster observations get relatively
+#'                            smaller weight than others in training the forest, i.e., the contribution
+#'                            of a given cluster to the final forest scales with the minimum of
+#'                            the number of observations in the cluster and samples_per_cluster.
+#' @param tune.parameters If true, NULL parameters are tuned by cross-validation; if false
+#'                        NULL parameters are set to defaults.
+#' @param num.fit.trees The number of trees in each 'mini forest' used to fit the tuning model.
+#' @param num.fit.reps The number of forests used to fit the tuning model.
+#' @param num.optimize.reps The number of random parameter values considered when using the model
+#'                          to select the optimal parameters.
+#' @param num.steps The number of boosting iterations. If NULL, selected by cross-validation
+#' @param tolerance If num.steps is NULL, the percentage of previous steps' error that must be estimated
+#'                  by cross validation in order to take a new step, default 0.95
+#' @param max.steps The maximum number of boosting iterations to try when num.steps NULL
+#' @param num.trees.tune If num.steps is NULL, the number of trees used to test a new boosting step when tuning
+#'        num.steps
+#'
+#' @return A trained regression forest object.
```

Comment:
Let's update this description of the return value to let users know that they can access the individual boosted forests in `boosted.forest$forests`, and also the debiased errors at each step.